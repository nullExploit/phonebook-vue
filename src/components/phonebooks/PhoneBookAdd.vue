<script setup>
import { usePhoneBookStore } from "@/stores/phonebook";
import { ref } from "vue";

const phonebookStore = usePhoneBookStore();
const name = ref("");
const phone = ref("");
const submit = (name, phone, router) => {
  if (name && phone) {
    phonebookStore.addPhoneBook(name, phone);
    return router.push("/");
  }
  router.push("/");
};
</script>

<template>
  <section>
    <div className="input">
      <input type="text" v-model="name" />
      <input
        type="text"
        v-model="phone"
        onkeypress="return event.charCode >= 48 && event.charCode <= 57"
      />
    </div>
    <div className="inputbutton">
      <button @click="() => submit(name, phone, $router)">save</button>
      <button @click="$router.push('/')">cancel</button>
    </div>
  </section>
</template>
