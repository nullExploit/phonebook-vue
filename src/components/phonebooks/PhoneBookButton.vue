<script setup>
import {
  faCheck,
  faFloppyDisk,
  faPenToSquare,
  faTrashCan,
  faXmark,
} from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";

defineProps({
  id: { type: String, required: true },
  name: { type: String, required: true },
  phone: { type: String, required: true },
  edit: { type: Boolean, required: true },
  remove: { type: Boolean, required: true },
});
</script>

<template>
  <div className="button" v-if="remove">
    <span className="buttons" @click="() => $emit('onDelete', id)">
      <FontAwesomeIcon :icon="faCheck" />
    </span>
    <span className="buttons" @click="() => $emit('onDeleteStatus', remove)">
      <FontAwesomeIcon :icon="faXmark" />
    </span>
  </div>
  <div className="button" v-else-if="edit">
    <span className="buttons" @click="() => $emit('onEdit', id, name, phone)">
      <FontAwesomeIcon aria-label="editbutton" :icon="faFloppyDisk" />
    </span>
  </div>
  <div className="button" v-else>
    <span className="buttons" @click="() => $emit('onEditStatus', edit)">
      <FontAwesomeIcon aria-label="editbutton" :icon="faPenToSquare" />
    </span>
    <span className="buttons" @click="() => $emit('onDeleteStatus', remove)">
      <FontAwesomeIcon :icon="faTrashCan" />
    </span>
  </div>
</template>
