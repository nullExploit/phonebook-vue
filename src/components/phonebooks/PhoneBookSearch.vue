<script setup>
import {
  faArrowDownAZ,
  faArrowUpZA,
  faMagnifyingGlass,
  faUserPlus,
} from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
defineProps({
  sort: { type: Boolean, required: true },
  limit: { type: Number, required: true },
});
</script>

<template>
  <div className="search-container">
    <button className="sorting-button" @click="$emit('changeSort', sort)">
      <FontAwesomeIcon :icon="!sort ? faArrowUpZA : faArrowDownAZ" />
    </button>
    <input
      className="search-bar"
      type="text"
      @keyup="(evt) => $emit('changeSearch', evt.target.value)"
    />
    <span>
      <FontAwesomeIcon :icon="faMagnifyingGlass" />
    </span>
    <button className="add-button" @click="$router.push('/add')">
      <FontAwesomeIcon :icon="faUserPlus" />
    </button>
  </div>
</template>
