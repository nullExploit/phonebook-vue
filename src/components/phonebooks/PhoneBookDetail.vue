<script setup>
import { ref } from "vue";
import PhoneBookButton from "./PhoneBookButton.vue";
import { usePhoneBookStore } from "@/stores/phonebook";

const props = defineProps({
  id: { type: String, required: true },
  name: { type: String, required: true },
  phone: { type: String, required: true },
});

const phonebookStore = usePhoneBookStore();
const edit = ref(false);
const remove = ref(false);
const tempName = ref(props.name);
const tempPhone = ref(props.phone);

const editStatus = (data) => {
  edit.value = !data;
};

const delStatus = (data) => {
  remove.value = !data;
};

const del = (id) => {
  phonebookStore.removePhoneBook(id);
};

const update = (id, name, phone) => {
  phonebookStore.updatePhoneBook(id, name, phone);
  edit.value = false;
};
</script>

<template>
  <div className="details" v-if="edit">
    <input type="text" className="edits" v-model="tempName" />
    <input
      type="text"
      className="edits"
      v-model="tempPhone"
      onkeypress="return event.charCode >= 48 && event.charCode <= 57"
    />
    <br />
    <PhoneBookButton
      :id="id"
      :name="tempName"
      :phone="tempPhone"
      :edit="edit"
      :remove="remove"
      @on-delete-status="delStatus"
      @on-delete="del"
      @on-edit-status="editStatus"
      @on-edit="update"
    />
  </div>
  <div className="details" v-else>
    <p className="name">{{ name }}</p>
    <p>{{ phone }}</p>
    <br />
    <PhoneBookButton
      :id="id"
      :name="tempName"
      :phone="tempPhone"
      :edit="edit"
      :remove="remove"
      @on-delete-status="delStatus"
      @on-delete="del"
      @on-edit-status="editStatus"
      @on-edit="update"
    />
  </div>
</template>
