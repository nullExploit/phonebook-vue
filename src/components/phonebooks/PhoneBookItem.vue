<script setup>
import { ref } from "vue";
import PhoneBookDetail from "./PhoneBookDetail.vue";
import { usePhoneBookStore } from "@/stores/phonebook";

defineProps({
  id: { type: String, required: true },
  name: { type: String, required: true },
  phone: { type: String, required: true },
  avatar: { type: String },
});

const phonebookStore = usePhoneBookStore();

const fileName = ref(null);
const file = ref(null);

const handleSubmit = async (event, id) => {
  event.preventDefault();
  phonebookStore.updatePhoneBookAvatar(id, file.value);
};

const handleChange = (event, id) => {
  if (event.target.files[0]) {
    fileName.value = URL.createObjectURL(event.target.files[0]);
    file.value = event.target.files[0];
    setTimeout(() => {
      document.getElementById(`submit${id}`).click();
    });
  }
};

const handleInput = (id) => {
  document.getElementById(`fileUpload${id}`).click();
};
</script>

<template>
  <div className="phonebook-item">
    <img
      className="img"
      :src="fileName ? fileName : avatar"
      :alt="file"
      @click="() => handleInput(id)"
    />
    <form
      @submit.prevent="(evt) => handleSubmit(evt, id)"
      :id="`formUpload${id}`"
      :style="{ display: 'none' }"
      encType="multipart/form-data"
    >
      <input
        type="file"
        :id="`fileUpload${id}`"
        accept="image/*"
        @change="(evt) => handleChange(evt, id)"
      />
      <button type="submit" :id="`submit${id}`" />
    </form>
    <PhoneBookDetail :name="name" :phone="phone" :id="id" />
  </div>
</template>
