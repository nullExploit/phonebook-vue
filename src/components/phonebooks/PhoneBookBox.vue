<script setup>
import { onMounted, onUpdated, ref } from "vue";
import PhoneBookSearch from "./PhoneBookSearch.vue";
import { usePhoneBookStore } from "@/stores/phonebook";
import PhoneBookItem from "./PhoneBookItem.vue";
import avatar from "@/assets/avatar.png";

const phonebookStore = usePhoneBookStore();

const sort = ref(false);
const search = ref("");
const page = ref(1);

const onScroll = () => {
  window.onscroll = () => {
    if (
      page.value < phonebookStore.total &&
      window.scrollY + window.innerHeight ===
        document.documentElement.scrollHeight
    ) {
      page.value += 1;
      phonebookStore.loadPhoneBook(sort.value, search.value, page.value);
    }
  };
};

const changeSearch = (data) => {
  search.value = data;
  page.value = 1;
  phonebookStore.loadPhoneBook(sort.value, search.value, page.value);
};

const changeSort = (data) => {
  sort.value = !data;
  page.value = 1;
  document.documentElement.scrollTop = 0;
  phonebookStore.loadPhoneBook(sort.value, search.value, page.value);
};

onMounted(() => {
  phonebookStore.loadPhoneBook(sort.value, search.value, page.value);
  onScroll();
});
</script>

<template>
  <section>
    <PhoneBookSearch
      :sort="sort"
      @change-search="changeSearch"
      @change-Sort="changeSort"
    />
    <div className="phonebooks">
      <PhoneBookItem
        v-for="item in phonebookStore.phonebooks"
        :id="item.id"
        :name="item.name"
        :phone="item.phone"
        :avatar="item.avatar ? `http://localhost:3001/${item.avatar}` : avatar"
        :key="item.id"
      />
    </div>
  </section>
</template>
