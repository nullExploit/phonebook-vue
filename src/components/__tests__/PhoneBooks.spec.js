import { describe, it, expect, vi, beforeAll } from "vitest";
import { mount } from "@vue/test-utils";
import { createTestingPinia } from "@pinia/testing";
import PhoneBookButton from "../phonebooks/PhoneBookButton.vue";
import PhoneBookItem from "../phonebooks/PhoneBookItem.vue";
import PhoneBookDetail from "../phonebooks/PhoneBookDetail.vue";
import PhoneBookSearch from "../phonebooks/PhoneBookSearch.vue";
import PhoneBookAdd from "../phonebooks/PhoneBookAdd.vue";
import PhoneBookBox from "../phonebooks/PhoneBookBox.vue";

describe("PhoneBooks", () => {
  beforeAll(() => {
    createTestingPinia({
      createSpy: vi.fn,
    });
  });

  it("Should render phonebook button correctly", async () => {
    const data = mount(PhoneBookButton, {
      props: {
        id: "123",
        name: "TESTING",
        phone: "081111111",
        remove: false,
        edit: false,
      },
    });

    expect(data.html()).toMatchSnapshot();
    expect(data.props().id).toBe("123");
    expect(data.props().name).toBe("TESTING");
    expect(data.props().phone).toBe("081111111");
    expect(data.props().remove).toBe(false);
    expect(data.props().edit).toBe(false);
    expect(
      data
        .findAll(".buttons")[0]
        .element.firstElementChild.getAttribute("data-icon")
    ).toBe("pen-to-square");
    expect(
      data
        .findAll(".buttons")[1]
        .element.firstElementChild.getAttribute("data-icon")
    ).toBe("trash-can");
    await data.setProps({ edit: true });
    expect(
      data.get(".buttons").element.firstElementChild.getAttribute("data-icon")
    ).toBe("floppy-disk");
    await data.setProps({ edit: false, remove: true });
    expect(
      data
        .findAll(".buttons")[0]
        .element.firstElementChild.getAttribute("data-icon")
    ).toBe("check");
    expect(
      data
        .findAll(".buttons")[1]
        .element.firstElementChild.getAttribute("data-icon")
    ).toBe("xmark");
    await data.setProps({
      id: "1",
      name: "TESTING DONE",
      phone: "081123",
      remove: false,
    });
    expect(data.props().id).toBe("1");
    expect(data.props().name).toBe("TESTING DONE");
    expect(data.props().phone).toBe("081123");
    expect(data.props().remove).toBe(false);
    expect(data.props().edit).toBe(false);
  });

  it("Should render phonebook detail correctly", () => {
    const data = mount(PhoneBookDetail, {
      props: {
        id: "123",
        name: "TESTING",
        phone: "08123123",
      },
    });

    expect(data.html()).toMatchSnapshot();
    expect(data.props().id).toBe("123");
    expect(data.props().name).toBe("TESTING");
    expect(data.props().phone).toBe("08123123");
  });

  it("Should render phonebook item correctly", async () => {
    const data = mount(PhoneBookItem, {
      props: {
        id: "123",
        name: "TESTING",
        phone: "08111111",
      },
    });

    expect(data.html()).toMatchSnapshot();
    expect(data.props().id).toBe("123");
    expect(data.props().name).toBe("TESTING");
    expect(data.props().phone).toBe("08111111");
    await data.findAll(".buttons")[0].trigger("click");
    expect(
      data
        .findAll(".buttons")[0]
        .element.firstElementChild.getAttribute("data-icon")
    ).toBe("floppy-disk");
    expect(data.findAll(".edits")[0].element.value).toBe("TESTING");
    expect(data.findAll(".edits")[1].element.value).toBe("08111111");
    expect(data.findAll(".edits")[0].element.tagName).toBe("INPUT");
    expect(data.findAll(".edits")[1].element.tagName).toBe("INPUT");
    await data.findAll(".buttons")[0].trigger("click");
    expect(
      data
        .findAll(".buttons")[0]
        .element.firstElementChild.getAttribute("data-icon")
    ).toBe("pen-to-square");
    expect(
      data
        .findAll(".buttons")[1]
        .element.firstElementChild.getAttribute("data-icon")
    ).toBe("trash-can");
    await data.findAll(".buttons")[1].trigger("click");
    expect(
      data
        .findAll(".buttons")[0]
        .element.firstElementChild.getAttribute("data-icon")
    ).toBe("check");
    expect(
      data
        .findAll(".buttons")[1]
        .element.firstElementChild.getAttribute("data-icon")
    ).toBe("xmark");
    await data.findAll(".buttons")[1].trigger("click");
  });

  it("Should render phonebook search correctly", async () => {
    const data = mount(PhoneBookSearch, {
      props: {
        sort: false,
        limit: 0,
      },
    });

    expect(data.html()).toMatchSnapshot();
    expect(data.props().sort).toBe(false);
    expect(data.props().limit).toBe(0);
    expect(
      data
        .get(".sorting-button")
        .element.firstElementChild.getAttribute("data-icon")
    ).toBe("arrow-up-z-a");
    expect(
      data
        .get("input")
        .element.nextElementSibling.firstElementChild.getAttribute("data-icon")
    ).toBe("magnifying-glass");
    expect(
      data
        .get(".add-button")
        .element.firstElementChild.getAttribute("data-icon")
    ).toBe("user-plus");
    await data.get("input").setValue("TESTING");
    expect(data.get("input").element.value).toBe("TESTING");
  });

  it("Should render phonebook add correctly", async () => {
    const data = mount(PhoneBookAdd);

    expect(data.html()).toMatchSnapshot();
    await data.findAll("input")[0].setValue("TESTING ADD");
    await data.findAll("input")[1].setValue("08123123123");
    expect(data.findAll("input")[0].element.value).toBe("TESTING ADD");
    expect(data.findAll("input")[1].element.value).toBe("08123123123");
  });

  it("Should render phonebook box correctly", async () => {
    const data = mount(PhoneBookBox);

    expect(data.html()).toMatchSnapshot();
    expect(
      data
        .get(".sorting-button")
        .element.firstElementChild.getAttribute("data-icon")
    ).toBe("arrow-up-z-a");
    await data.get(".sorting-button").trigger("click");
    expect(
      data
        .get(".sorting-button")
        .element.firstElementChild.getAttribute("data-icon")
    ).toBe("arrow-down-a-z");
    await data.get(".sorting-button").trigger("click");
    await data.get(".search-bar").setValue("TESTING");
    expect(data.get(".search-bar").element.value).toBe("TESTING");
  });
});
