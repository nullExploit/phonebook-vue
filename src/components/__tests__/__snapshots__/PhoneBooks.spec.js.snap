// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PhoneBooks > Should render phonebook add correctly 1`] = `
"<section>
  <div class="input"><input type="text"><input type="text" onkeypress="return event.charCode >= 48 &amp;&amp; event.charCode <= 57"></div>
  <div class="inputbutton"><button>save</button><button>cancel</button></div>
</section>"
`;

exports[`PhoneBooks > Should render phonebook box correctly 1`] = `
"<section>
  <div class="search-container"><button class="sorting-button"><svg class="svg-inline--fa fa-arrow-up-z-a" aria-hidden="true" focusable="false" data-prefix="fas" data-icon="arrow-up-z-a" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 576 512">
        <path class="" fill="currentColor" d="M183.6 42.4C177.5 35.8 169 32 160 32s-17.5 3.8-23.6 10.4l-88 96c-11.9 13-11.1 33.3 2 45.2s33.3 11.1 45.2-2L128 146.3 128 448c0 17.7 14.3 32 32 32s32-14.3 32-32l0-301.7 32.4 35.4c11.9 13 32.2 13.9 45.2 2s13.9-32.2 2-45.2l-88-96zM320 64c0 17.7 14.3 32 32 32l50.7 0-73.4 73.4c-9.2 9.2-11.9 22.9-6.9 34.9s16.6 19.8 29.6 19.8l128 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l-50.7 0 73.4-73.4c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L352 32c-17.7 0-32 14.3-32 32zm96 192c-12.1 0-23.2 6.8-28.6 17.7l-64 128-16 32c-7.9 15.8-1.5 35 14.3 42.9s35 1.5 42.9-14.3l7.2-14.3 88.4 0 7.2 14.3c7.9 15.8 27.1 22.2 42.9 14.3s22.2-27.1 14.3-42.9l-16-32-64-128C439.2 262.8 428.1 256 416 256zM395.8 400L416 359.6 436.2 400l-40.4 0z"></path>
      </svg></button><input class="search-bar" type="text"><span><svg class="svg-inline--fa fa-magnifying-glass" aria-hidden="true" focusable="false" data-prefix="fas" data-icon="magnifying-glass" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path class="" fill="currentColor" d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"></path></svg></span><button class="add-button"><svg class="svg-inline--fa fa-user-plus" aria-hidden="true" focusable="false" data-prefix="fas" data-icon="user-plus" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 640 512">
        <path class="" fill="currentColor" d="M96 128a128 128 0 1 1 256 0A128 128 0 1 1 96 128zM0 482.3C0 383.8 79.8 304 178.3 304l91.4 0C368.2 304 448 383.8 448 482.3c0 16.4-13.3 29.7-29.7 29.7L29.7 512C13.3 512 0 498.7 0 482.3zM504 312l0-64-64 0c-13.3 0-24-10.7-24-24s10.7-24 24-24l64 0 0-64c0-13.3 10.7-24 24-24s24 10.7 24 24l0 64 64 0c13.3 0 24 10.7 24 24s-10.7 24-24 24l-64 0 0 64c0 13.3-10.7 24-24 24s-24-10.7-24-24z"></path>
      </svg></button></div>
  <div class="phonebooks"></div>
</section>"
`;

exports[`PhoneBooks > Should render phonebook button correctly 1`] = `"<div class="button"><span class="buttons"><svg class="svg-inline--fa fa-pen-to-square" aria-hidden="true" focusable="false" data-prefix="fas" data-icon="pen-to-square" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512" aria-label="editbutton"><path class="" fill="currentColor" d="M471.6 21.7c-21.9-21.9-57.3-21.9-79.2 0L362.3 51.7l97.9 97.9 30.1-30.1c21.9-21.9 21.9-57.3 0-79.2L471.6 21.7zm-299.2 220c-6.1 6.1-10.8 13.6-13.5 21.9l-29.6 88.8c-2.9 8.6-.6 18.1 5.8 24.6s15.9 8.7 24.6 5.8l88.8-29.6c8.2-2.7 15.7-7.4 21.9-13.5L437.7 172.3 339.7 74.3 172.4 241.7zM96 64C43 64 0 107 0 160L0 416c0 53 43 96 96 96l256 0c53 0 96-43 96-96l0-96c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 96c0 17.7-14.3 32-32 32L96 448c-17.7 0-32-14.3-32-32l0-256c0-17.7 14.3-32 32-32l96 0c17.7 0 32-14.3 32-32s-14.3-32-32-32L96 64z"></path></svg></span><span class="buttons"><svg class="svg-inline--fa fa-trash-can" aria-hidden="true" focusable="false" data-prefix="fas" data-icon="trash-can" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path class="" fill="currentColor" d="M135.2 17.7C140.6 6.8 151.7 0 163.8 0L284.2 0c12.1 0 23.2 6.8 28.6 17.7L320 32l96 0c17.7 0 32 14.3 32 32s-14.3 32-32 32L32 96C14.3 96 0 81.7 0 64S14.3 32 32 32l96 0 7.2-14.3zM32 128l384 0 0 320c0 35.3-28.7 64-64 64L96 512c-35.3 0-64-28.7-64-64l0-320zm96 64c-8.8 0-16 7.2-16 16l0 224c0 8.8 7.2 16 16 16s16-7.2 16-16l0-224c0-8.8-7.2-16-16-16zm96 0c-8.8 0-16 7.2-16 16l0 224c0 8.8 7.2 16 16 16s16-7.2 16-16l0-224c0-8.8-7.2-16-16-16zm96 0c-8.8 0-16 7.2-16 16l0 224c0 8.8 7.2 16 16 16s16-7.2 16-16l0-224c0-8.8-7.2-16-16-16z"></path></svg></span></div>"`;

exports[`PhoneBooks > Should render phonebook detail correctly 1`] = `
"<div class="details">
  <p class="name">TESTING</p>
  <p>08123123</p><br>
  <div class="button"><span class="buttons"><svg class="svg-inline--fa fa-pen-to-square" aria-hidden="true" focusable="false" data-prefix="fas" data-icon="pen-to-square" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512" aria-label="editbutton"><path class="" fill="currentColor" d="M471.6 21.7c-21.9-21.9-57.3-21.9-79.2 0L362.3 51.7l97.9 97.9 30.1-30.1c21.9-21.9 21.9-57.3 0-79.2L471.6 21.7zm-299.2 220c-6.1 6.1-10.8 13.6-13.5 21.9l-29.6 88.8c-2.9 8.6-.6 18.1 5.8 24.6s15.9 8.7 24.6 5.8l88.8-29.6c8.2-2.7 15.7-7.4 21.9-13.5L437.7 172.3 339.7 74.3 172.4 241.7zM96 64C43 64 0 107 0 160L0 416c0 53 43 96 96 96l256 0c53 0 96-43 96-96l0-96c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 96c0 17.7-14.3 32-32 32L96 448c-17.7 0-32-14.3-32-32l0-256c0-17.7 14.3-32 32-32l96 0c17.7 0 32-14.3 32-32s-14.3-32-32-32L96 64z"></path></svg></span><span class="buttons"><svg class="svg-inline--fa fa-trash-can" aria-hidden="true" focusable="false" data-prefix="fas" data-icon="trash-can" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path class="" fill="currentColor" d="M135.2 17.7C140.6 6.8 151.7 0 163.8 0L284.2 0c12.1 0 23.2 6.8 28.6 17.7L320 32l96 0c17.7 0 32 14.3 32 32s-14.3 32-32 32L32 96C14.3 96 0 81.7 0 64S14.3 32 32 32l96 0 7.2-14.3zM32 128l384 0 0 320c0 35.3-28.7 64-64 64L96 512c-35.3 0-64-28.7-64-64l0-320zm96 64c-8.8 0-16 7.2-16 16l0 224c0 8.8 7.2 16 16 16s16-7.2 16-16l0-224c0-8.8-7.2-16-16-16zm96 0c-8.8 0-16 7.2-16 16l0 224c0 8.8 7.2 16 16 16s16-7.2 16-16l0-224c0-8.8-7.2-16-16-16zm96 0c-8.8 0-16 7.2-16 16l0 224c0 8.8 7.2 16 16 16s16-7.2 16-16l0-224c0-8.8-7.2-16-16-16z"></path></svg></span></div>
</div>"
`;

exports[`PhoneBooks > Should render phonebook item correctly 1`] = `
"<div class="phonebook-item"><img class="img">
  <form id="formUpload123" style="display: none;" enctype="multipart/form-data"><input type="file" id="fileUpload123" accept="image/*"><button type="submit" id="submit123"></button></form>
  <div class="details">
    <p class="name">TESTING</p>
    <p>08111111</p><br>
    <div class="button"><span class="buttons"><svg class="svg-inline--fa fa-pen-to-square" aria-hidden="true" focusable="false" data-prefix="fas" data-icon="pen-to-square" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512" aria-label="editbutton"><path class="" fill="currentColor" d="M471.6 21.7c-21.9-21.9-57.3-21.9-79.2 0L362.3 51.7l97.9 97.9 30.1-30.1c21.9-21.9 21.9-57.3 0-79.2L471.6 21.7zm-299.2 220c-6.1 6.1-10.8 13.6-13.5 21.9l-29.6 88.8c-2.9 8.6-.6 18.1 5.8 24.6s15.9 8.7 24.6 5.8l88.8-29.6c8.2-2.7 15.7-7.4 21.9-13.5L437.7 172.3 339.7 74.3 172.4 241.7zM96 64C43 64 0 107 0 160L0 416c0 53 43 96 96 96l256 0c53 0 96-43 96-96l0-96c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 96c0 17.7-14.3 32-32 32L96 448c-17.7 0-32-14.3-32-32l0-256c0-17.7 14.3-32 32-32l96 0c17.7 0 32-14.3 32-32s-14.3-32-32-32L96 64z"></path></svg></span><span class="buttons"><svg class="svg-inline--fa fa-trash-can" aria-hidden="true" focusable="false" data-prefix="fas" data-icon="trash-can" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path class="" fill="currentColor" d="M135.2 17.7C140.6 6.8 151.7 0 163.8 0L284.2 0c12.1 0 23.2 6.8 28.6 17.7L320 32l96 0c17.7 0 32 14.3 32 32s-14.3 32-32 32L32 96C14.3 96 0 81.7 0 64S14.3 32 32 32l96 0 7.2-14.3zM32 128l384 0 0 320c0 35.3-28.7 64-64 64L96 512c-35.3 0-64-28.7-64-64l0-320zm96 64c-8.8 0-16 7.2-16 16l0 224c0 8.8 7.2 16 16 16s16-7.2 16-16l0-224c0-8.8-7.2-16-16-16zm96 0c-8.8 0-16 7.2-16 16l0 224c0 8.8 7.2 16 16 16s16-7.2 16-16l0-224c0-8.8-7.2-16-16-16zm96 0c-8.8 0-16 7.2-16 16l0 224c0 8.8 7.2 16 16 16s16-7.2 16-16l0-224c0-8.8-7.2-16-16-16z"></path></svg></span></div>
  </div>
</div>"
`;

exports[`PhoneBooks > Should render phonebook search correctly 1`] = `
"<div class="search-container"><button class="sorting-button"><svg class="svg-inline--fa fa-arrow-up-z-a" aria-hidden="true" focusable="false" data-prefix="fas" data-icon="arrow-up-z-a" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 576 512">
      <path class="" fill="currentColor" d="M183.6 42.4C177.5 35.8 169 32 160 32s-17.5 3.8-23.6 10.4l-88 96c-11.9 13-11.1 33.3 2 45.2s33.3 11.1 45.2-2L128 146.3 128 448c0 17.7 14.3 32 32 32s32-14.3 32-32l0-301.7 32.4 35.4c11.9 13 32.2 13.9 45.2 2s13.9-32.2 2-45.2l-88-96zM320 64c0 17.7 14.3 32 32 32l50.7 0-73.4 73.4c-9.2 9.2-11.9 22.9-6.9 34.9s16.6 19.8 29.6 19.8l128 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l-50.7 0 73.4-73.4c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L352 32c-17.7 0-32 14.3-32 32zm96 192c-12.1 0-23.2 6.8-28.6 17.7l-64 128-16 32c-7.9 15.8-1.5 35 14.3 42.9s35 1.5 42.9-14.3l7.2-14.3 88.4 0 7.2 14.3c7.9 15.8 27.1 22.2 42.9 14.3s22.2-27.1 14.3-42.9l-16-32-64-128C439.2 262.8 428.1 256 416 256zM395.8 400L416 359.6 436.2 400l-40.4 0z"></path>
    </svg></button><input class="search-bar" type="text"><span><svg class="svg-inline--fa fa-magnifying-glass" aria-hidden="true" focusable="false" data-prefix="fas" data-icon="magnifying-glass" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path class="" fill="currentColor" d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"></path></svg></span><button class="add-button"><svg class="svg-inline--fa fa-user-plus" aria-hidden="true" focusable="false" data-prefix="fas" data-icon="user-plus" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 640 512">
      <path class="" fill="currentColor" d="M96 128a128 128 0 1 1 256 0A128 128 0 1 1 96 128zM0 482.3C0 383.8 79.8 304 178.3 304l91.4 0C368.2 304 448 383.8 448 482.3c0 16.4-13.3 29.7-29.7 29.7L29.7 512C13.3 512 0 498.7 0 482.3zM504 312l0-64-64 0c-13.3 0-24-10.7-24-24s10.7-24 24-24l64 0 0-64c0-13.3 10.7-24 24-24s24 10.7 24 24l0 64 64 0c13.3 0 24 10.7 24 24s-10.7 24-24 24l-64 0 0 64c0 13.3-10.7 24-24 24s-24-10.7-24-24z"></path>
    </svg></button></div>"
`;
